analytics/policy/services/registration_number_policies.py

```diff
@@ -0,0 +1,39 @@
+import re
+import json
+import urllib3
+
+class RegistrationNumberPolicies:
+    def __init__(self, registration_number):
+        self.registration_number = registration_number
+        self.session = urllib3.PoolManager()
+
+    def valid_pattern(self):
+        valid_registration_pattern = re.compile(r'^[0-9]{2}-[0-9]{6}$')
+
+        if valid_registration_pattern.match(self.registration_number):
+            return True
+        else:
+            return False
+
+    def valid_registration_number(self):
+        facet_name = "licencenumber"
+        limit_value = 20
+        OPEN_DATA_VANCOUVER_URL = (
+            "https://opendata.vancouver.ca/"
+            "api/explore/v2.1/catalog/datasets/"
+            "business-licences/records?"
+            f"limit={limit_value}&"
+            f"refine={facet_name}%3A{self.registration_number}"
+        )
+        response = self.session.request(
```

Comment:
There's no need to access the API as this is handled in the API client, please check PR #38 (this is still under review, so let me know if you have any concerns). You can call the client here
The return values from the client will be in the form of ["status"], ["status1", "status2"], ... Please check the API client unit test in the same PR for examples

Reply:
with respect to Yang's comment, you can make it like:

ValidRegistrationNumberPolicy(Policy)
UniqueRegistrationNumberPolicy(Policy)
ExistedRegistrationNumberPolicy(Policy)

where Policy is the interface with any shared method/attribute
These Policy objects mimic the strategy design pattern